ui/src/app/views/workflow/run/node/artifact/artifact.list.component.ts

```diff
@@ -14,14 +14,18 @@ export class WorkflowRunArtifactListComponent extends Table {
 
     // Allow angular update from work started outside angular context
     zone: NgZone;
+    filter: string;
 
     constructor() {
         super();
         this.zone = new NgZone({enableLongStackTrace: false});
     }
 
     getData(): any[] {
-        return this.artifacts;
+        if (!this.filter) {
+            return this.artifacts;
+        }
+        return this.artifacts.filter(v => (v.name.indexOf(this.filter) !== -1 || v.sha512sum.indexOf(this.filter) !== -1));
```

Comment:
I think in case of sha512 research it's better to use `===` instead of `indexOf`